lang/src/org/partiql/lang/eval/visitors/SelectStarVisitorTransform.kt

```diff
@@ -0,0 +1,117 @@
+package org.partiql.lang.eval.visitors
+
+import com.amazon.ionelement.api.emptyMetaContainer
+import org.partiql.lang.ast.UniqueNameMeta
+import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.eval.errNoContext
+
+class SelectStarVisitorTransform : PartiqlAst.VisitorTransform() {
+
+    /**
+     * Copies all parts of [PartiqlAst.Expr.Select] except [newProjection] for [PartiqlAst.Projection].
+     */
+    private fun copyProjectionToSelect(node: PartiqlAst.Expr.Select, newProjection: PartiqlAst.Projection): PartiqlAst.Expr {
+        return PartiqlAst.build {
+            select(
+                setq = node.setq,
+                project = newProjection,
+                from = node.from,
+                fromLet = node.fromLet,
+                where = node.where,
+                group = node.group,
+                having = node.having,
+                limit = node.limit,
+                metas = node.metas)
+        }
+    }
+
+    override fun transformExprSelect(node: PartiqlAst.Expr.Select): PartiqlAst.Expr {
+        val rewritten = super.transformExprSelect(node) as PartiqlAst.Expr.Select
+
+        val projection = rewritten.project
+
+        // Check if SELECT * is being used.
+        if (projection is PartiqlAst.Projection.ProjectStar) {
```

Comment:
This check is slightly different than the ExprNode rewriter version. In that version, a `SelectProjection` could be a `SelectProjectionList`, which has a list of `SelectListItem`. Of these items, it can be a `SelectListItemExpr`, `SelectListItemProjectAll`, or **`SelectListItemStar`**.

While in PIG, a `ProjectStar` (~ `SelectListItemStar`) can directly be a `Projection` (~ `SelectProjection`). So not sure if we need the checks from the original [code](https://github.com/partiql/partiql-lang-kotlin/blob/80e8665bd167e98ad7967eb4fda096948c18a028/lang/src/org/partiql/lang/ast/passes/SelectStarRewriter.kt#L27).

Reply:
Yes, the `PartiqlAst` models `SELECT *` better than `ExprNode` does...  The result of the better modeling is that those checks are in fact not needed here.